### DISCUSSION TITLE
Maybe Bug: Alpine & Livewire causing loss of events

### DISCUSSION THREAD

**Discussion by Peter-Fogarty** [alpinejs/alpine#3173] [1. Help] [answered]
When using the JetStream modals that come with Laravel starter, in the User profile for example, if I press the button to delete a user and then press cancel instead, then I can no longer click on anything else on the page, I must reload before it listens to click events again. I am sure it is a dependency clash or something like that, but I can't figure it out. I do get the following in the browser console though, which shows its deep in the bowels of Alpine,
```
[Error] Unhandled Promise Rejection: [object Object]
	(anonymous function)
	rejectPromise
	(anonymous function) (cdn.min.js:5:8676)
	(anonymous function) (alpinejs.js:1058)
	(anonymous function) (alpinejs.js:835)
	transition (alpinejs.js:1008)
	out (alpinejs.js:948)
	(anonymous function) (cdn.min.js:5:8621)
	Promise
	(anonymous function) (cdn.min.js:5:8593)
	(anonymous function) (cdn.min.js:5:30592)
	(anonymous function) (cdn.min.js:5:6120)
	(anonymous function) (cdn.min.js:5:30683)
	we (cdn.min.js:5:1284)
	(anonymous function) (cdn.min.js:5:1026)
	Yt (cdn.min.js:1:4880)
	Yt
	(anonymous function) (cdn.min.js:5:30628)
	r (cdn.min.js:5:16267)
	ln (cdn.min.js:1:259)
[Error] Unhandled Promise Rejection: TypeError: u is not a function. (In 'u()', 'u' is undefined)
	(anonymous function) (cdn.min.js:5:9054)
	promiseReactionJob
```

Any ideas?

**Comment by ekwoka** [comment]
I don't use Laravel, but it could be helpful to use the non-minified Alpine to see more info about where in Alpine it's going wrong.

**Reply by Peter-Fogarty** [reply]
Good suggestion, I can now report that the Promise rejection is in x-transition.js, line 262...
```
el._x_transitioning = {
        beforeCancels: [],
        beforeCancel(callback) { this.beforeCancels.push(callback) },
        cancel: once(function () { while (this.beforeCancels.length) { this.beforeCancels.shift()() }; finish(); }),
        finish,
    }
```
Specifically, it is the --> this.beforeCancels.shift()()
part that is being highlighted as in error.....

I am not very good at JS, thats why I liked Alpine so much... haha... but it seems I need to learn more to get to the bottom of this.
The package I am using from laravel, is the most recent one in the github repo, so my brain tells me it must be some sneaky class of something, rather than a bug per se... I am wondering if it Vite, and what it did to my resource/js/app.js to start Alpine......

**Comment by Peter-Fogarty** [comment, marked as answer]
So I was working on the theory that something was clashing with Alpine... I was trying to work out what could clash with Alpine, and it turns out the answer was Alpine. I think as part of my migration to Vite, I ended up with the NPM install of Alpine, while my main project template was still pulling in the CDN version.... I removed the CDN script tag and everything started working again. So there you go. Problem solved.
